### PR TITLE
Transform prices inputs/outpus via DTO

### PR DESCRIPTION
**Pull request by brizyNicu**
Changes:
- Add Output Dto and transformers for Order, OrderItem, ChannelPricing;
- Add Input Dto and transformers for ChannelPricing;
- Add a new file to register data-transformer-services and registered it in ` ApiBundle's` Extension

**Related Issues**:
- bagrinsergiu/brizy-cms-ui#2238
- bagrinsergiu/brizy-cloud-sylius-demo-template/issues/178
- bagrinsergiu/brizy-cloud-sylius-demo-template/issues/191

### CODE REVIEW

**Review by DanxilL — CHANGES_REQUESTED** [review]

File: src/Sylius/Bundle/ApiBundle/DataTransformer/OrderItemReadDataTransformer.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        $output = new OrderItemRead();
+        $output->variant = $object->getVariant();
+        $output->productName = $object->getProductName();
+        $output->id = $object->getId();
+        $output->order = $object->getOrder();
+        $output->quantity = $object->getQuantity();
+        $output->units = $object->getUnits();
+        $output->adjustments = $object->getAdjustments();
+//        $output->product = $object->getProduct();
```
[the remark this answers is not in the document]
Comment (brizyNicu) [reply src/Sylius/Bundle/ApiBundle/DataTransformer/OrderItemReadDataTransformer.php:28]: outdated

File: src/Sylius/Bundle/ApiBundle/Helpers/PriceTransformer.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        return $value / self::PRICE_DIVIDER;
+    }
+
+    public static function unitsToCentimes(?float $value)
+    {
+        if (!$value) {
+            return null;
+        }
+        return $value * self::PRICE_MULTIPLIER;
+    }
```
[the remark this answers is not in the document]
Comment (brizyNicu) [reply src/Sylius/Bundle/ApiBundle/Helpers/PriceTransformer.php:24]: outdated

File: src/Sylius/Bundle/ApiBundle/DataTransformer/ChannelPricingInputDataTransformer.php [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        $channelPricing->setPrice(PriceTransformer::unitsToCentimes($input->price));
+        $channelPricing->setOriginalPrice(PriceTransformer::unitsToCentimes($input->originalPrice));
+
+        return $channelPricing;
+    }
+
+    public function supportsTransformation($data, string $to, array $context = []): bool
+    { dd($data);
+        if ($data instanceof ChannelPricing) {
```
[the remark this answers is not in the document]
Comment (brizyNicu) [reply src/Sylius/Bundle/ApiBundle/DataTransformer/ChannelPricingInputDataTransformer.php:37]: removed

File: src/Sylius/Bundle/ApiBundle/DTO/OrderItemRead.php [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+     * @var ArrayCollection|Collection|OrderItemUnitInterface[]
+     */
+    public $units;
+
+//    /**
+//     * @ApiProperty(identifier=false)
+//     * @Groups({"admin:order_item:read", "shop:order_item:read"})
+//     */
+//    public ?ProductInterface $product;
+
```
[the remark this answers is not in the document]
Comment (brizyNicu) [reply src/Sylius/Bundle/ApiBundle/DTO/OrderItemRead.php:59]: outdated

**Pull request closed without merging**